io.c

```diff
@@ -70,7 +70,7 @@ uint32_t memory_read_str(memory_t *m, uint8_t *dst, uint32_t addr, uint32_t max)
 uint32_t memory_read_ifetch(memory_t *m, uint32_t addr)
 {
     const uint32_t addr_lo = addr & mask_lo;
-    assert((addr_lo & 3) == 0);
+    assert((addr_lo & 1) == 0);
```

Comment:
Why?

Reply:
After executing a compressed instruction, the Program Counter only plus 2, and we can't promise that the last two bits of address always remain 0.  Thus, we modified `(addr_lo & 3)` into `(addr_lo & 1)`.